sbi/inference/posteriors/vi_posterior.py

```diff
@@ -21,32 +21,36 @@
     make_object_deepcopy_compatible,
     move_all_tensor_to_device,
 )
-from sbi.types import Shape, TorchTransform
+from sbi.types import (
+    PyroTransformedDistribution,
+    Shape,
+    TorchDistribution,
+    TorchTensor,
```

Comment:
why do we have this extra type, and not just `Tensor`?

Reply:
When using the newly defined `TorchTensor`, it is displayed as `Optional[Tensor]`, which makes a big difference

Reply:
I see, makes sense. coudn't we call this new type `Tensor` as well, that would be even shorter and all we need to change throughout the code base is the import `from torch import Tensor` to `from sbi.types import Tensor`?

Reply:
Unfortunately, `pyright` does not understand that the `sbi.types.Tensor` is the same as `torch.Tensor` (so it complains if we mix and match them). So we would be allowed to **only** use `sbi.types.Tensor` if we want to do this. I think I would prefer to only use the `sbi.types.Tensor` where it is strictly needed and stick to `torch.Tensor` otherwise. What do you think?